src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -160,6 +161,23 @@ public static void writeRandomDecimals(int size, IonType type, String path, Stri
         WriteRandomIonValues.printInfo(path);
     }
 
+    /**
+     * Write random Ion decimals into the file, and all generated data is conformed with the constraints provided by Ion schema file.
+     * @param size specifies the size in bytes of the generated file.
+     * @param type determines which type of ion data will be generated.
```

Comment:
Is `type` implicit, since this method is for writing decimals? What happens if a non-decimal type is provided?

Reply:
> Is `type` implicit, since this method is for writing decimals? What happens if a non-decimal type is provided?

I think 'type' should be explicit parameter because it was predefined as 'DECIMAL' before calling this method. The reason passing the parameter 'type' is for the further usage of calling method  WriteRandomIonValues.writeRequestedSizeFile.

Reply:
Is it always DECIMAL when calling this method?

Reply:
> Is it always DECIMAL when calling this method?

Yes, it is always DECIMAL

Reply:
Then why does it need to be passed into this method? Can't this method just pass IonType.DECIMAL to the method it calls?